pkg/server/container_create.go

```diff
@@ -418,11 +418,20 @@ func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxP
 
 	setOCILinuxResource(&g, config.GetLinux().GetResources())
 
-	if sandboxConfig.GetLinux().GetCgroupParent() != "" {
```

Comment:
We should:
1) Either move implementation of these options to the place the option was originally configured, e.g. `OOMScoreAdj` is configured in [`setOCILinuxResource`](https://github.com/containerd/cri/blob/70da14e4b37909431afb86508d9d83d49136a98c/pkg/server/container_create.go#L755);
2) Or add a separate function to handle Rootless, and move all the implementation of these options there. And we should make sure the `Rootless` function is called after the spec generation, so that it can cleanup unwanted options.

For example, the apparmor clearing up here won't take effect, because it will be overwrite by [this](https://github.com/containerd/cri/blob/70da14e4b37909431afb86508d9d83d49136a98c/pkg/server/container_create.go#L235).

Reply:
I vote #1..